config/cmake/addBoost.cmake

```diff
@@ -61,170 +70,111 @@ if(WIN32 AND NOT UNIX_LIKE)
             endif()
         endforeach()
     endforeach()
-
+    message(STATUS "${boost_paths}")
     find_path(
         BOOST_TEST_PATH
         NAMES boost/version.hpp
         HINTS ENV BOOST_INSTALL_PATH
         PATHS ${BOOST_INSTALL_PATH} ${boost_paths}
     )
-
     if(BOOST_TEST_PATH)
-        set(Boost_ROOT ${BOOST_TEST_PATH})
-    endif(BOOST_TEST_PATH)
-else()
-    if(NOT Boost_ROOT)
-        if(BOOST_INSTALL_PATH)
-            set(Boost_ROOT "${BOOST_INSTALL_PATH}")
-        elseif($ENV{BOOST_INSTALL_PATH})
-            set(Boost_ROOT "$ENV{BOOST_INSTALL_PATH}")
+        if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.30)
+            find_path(
+                BOOST_CMAKE_PATH
+                NAMES BoostConfig.cmake
+                PATHS ${BOOST_TEST_PATH}/${BOOST_MSVC_LIB_PATH}/cmake
+                PATH_SUFFIXES
+                    Boost-1.86.0
+                    Boost-1.85.0
+                    Boost-1.84.0
+                    Boost-1.83.0
+                    Boost-1.82.0
+                    Boost-1.81.0
+                    Boost-1.80.0
+                    Boost-1.79.0
+                    Boost-1.78.0
+                    Boost-1.77.0
+                    Boost-1.76.0
+                    Boost-1.75.0
+            )
+            set(Boost_ROOT ${BOOST_CMAKE_PATH})
         else()
-            set(Boost_ROOT "$ENV{BOOST_ROOT}")
+            set(Boost_ROOT ${BOOST_TEST_PATH})
         endif()
+    endif(BOOST_TEST_PATH)
+endif()
+# Minimum version of Boost required for building a project
+
+if(NOT Boost_ROOT)
+    message(STATUS "Boost root not found ${Boost_ROOT}")
+    if(BOOST_INSTALL_PATH)
+        set(Boost_ROOT "${BOOST_INSTALL_PATH}")
+    elseif($ENV{BOOST_INSTALL_PATH})
+        set(Boost_ROOT "$ENV{BOOST_INSTALL_PATH}")
+    else()
+        set(Boost_ROOT "$ENV{BOOST_ROOT}")
     endif()
 endif()
 
+if($ENV{BOOST_ROOT})
+    set(BOOST_ROOT "$ENV{BOOST_ROOT}")
```

Comment:
Might be easier to follow if this new variable has a `_ENV` postfix added to the name, instead of just using the different case to distinguish it from `Boost_ROOT`.

Reply:
BOOST_ROOT and Boost_ROOT are standard variables that are both supposed to work and do the same thing

Reply:
Looking more into how find_package locates Boost, there could be some unexpected behavior finding Boost with CMake prior to 3.30 without a BoostConfig.cmake file (`Boost_NO_BOOST_CMAKE` set to `ON`). We might want to set BOOST_ROOT to mirror Boost_ROOT instead of ENV{BOOST_ROOT}.

---

For finding Boost using Config mode with find_package:
* Based on CMP0144, prior to 3.27 find_package ignores `<PACKAGENAME>_ROOT` variables, making this a no-op. For CMake >=3.27, supposedly find_package checks the `<PACKAGENAME>_ROOT` environment variable, so setting BOOST_ROOT to the environment variable is redundant, other than our find_path call on line 141 (which could use an alternate name for our temp variable or directly use `$ENV{BOOST_ROOT}`).

For finding Boost using the FindBoost module (only applies to CMake <3.30, see CMP0167):
* [FindBoost.cmake](https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake#L603) ignores BOOST_ROOT if Boost_ROOT is set, 
and has its own check for whatever is in $ENV{BOOST_ROOT} for trying to do a find_package call using CMake Config mode first.
* If the find_package check fails, it does a bunch of stuff to locate Boost -- the fallback search also uses the [environment variables](https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake#L1738) for possible locations, however it looks like it might only check the location given in [BOOST_ROOT but not Boost_ROOT](https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake#L1751). This could result in whatever location was given in `BOOST_INSTALL_PATH` not being searched at all.

Reply:
I will look into it a little more this evening

Reply:
I am changing BOOST_ROOT to mirror Boost_ROOT if it is not set